nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentManagerImpl.java

```diff
@@ -402,39 +386,52 @@ private static NuxeoPrincipal getAuthor(DocumentModel docModel) {
-        }
+        return doPrivilledged(comment, session -> {
+            Map<String, Object> ctxMap = Collections.singletonMap(ResourceAdapter.CORE_SESSION_CONTEXT_KEY, session);
+            RelationManager relationManager = Framework.getService(RelationManager.class);
+            Graph graph = relationManager.getGraph(config.graphName, session);
+            Resource commentResource = relationManager.getResource(config.commentNamespace, comment, ctxMap);
+            if (commentResource == null) {
+                throw new NuxeoException("Could not adapt document model to relation resource ; "
+                        + "check the service relation adapters configuration");
+            }
+            Resource predicate = new ResourceImpl(config.predicateNamespace);
+
+            List<Statement> statementList = graph.getStatements(commentResource, predicate, null);
+            if (graph instanceof JenaGraph) {
+                // XXX AT: BBB for when repository name was not included in the
+                // resource uri
+                Resource oldDocResource = new QNameResourceImpl(config.commentNamespace, comment.getId());
+                statementList.addAll(graph.getStatements(oldDocResource, predicate, null));
+            }
 
-        List<DocumentModel> docList = new ArrayList<DocumentModel>();
-        for (Statement stmt : statementList) {
-            QNameResourceImpl subject = (QNameResourceImpl) stmt.getObject();
-            DocumentModel docModel = (DocumentModel) relationManager.getResourceRepresentation(
-                    config.documentNamespace, subject, ctxMap);
-            if (docModel == null) {
-                log.warn("Could not adapt comment relation subject to a document "
-                        + "model; check the service relation adapters configuration");
-                continue;
+            List<DocumentModel> docList = new ArrayList<DocumentModel>();
+            for (Statement stmt : statementList) {
+                QNameResourceImpl subject = (QNameResourceImpl) stmt.getObject();
+                DocumentModel docModel = (DocumentModel) relationManager.getResourceRepresentation(
+                        config.documentNamespace, subject, ctxMap);
+                if (docModel == null) {
+                    log.warn("Could not adapt comment relation subject to a document "
+                            + "model; check the service relation adapters configuration");
+                    continue;
+                }
+                // detach the document as it was loaded by a system session, not the user session.
+                docModel.detach(true);
+                docList.add(docModel);
             }
-            docList.add(docModel);
-        }
-        return docList;
+            return docList;
+        });
+    }
 
+    /**
+     * @since 11.1
+     */
+    protected <S> S doPrivilledged(DocumentModel doc, Function<CoreSession, S> function) {
```

Comment:
A typo in name:
```suggestion
    protected <S> S doPrivileged(DocumentModel doc, Function<CoreSession, S> function) {
```
Furthermore, as Thomas pointed out to me, we don't need the generic type as both methods calling this one want to get a `List<DocumentModel>`.

Reply:
I agree about the generic type !